src/extension/api/codeEditor.ts

```diff
@@ -0,0 +1,31 @@
+import * as sourcegraph from 'sourcegraph'
+import { ClientCodeEditorAPI } from '../../client/api/codeEditor'
+import * as plain from '../../protocol/plainTypes'
+import { Range } from '../types/range'
+import { ExtDocuments } from './documents'
+
+/** @internal */
+export class ExtCodeEditor implements sourcegraph.CodeEditor {
+    constructor(private resource: string, private proxy: ClientCodeEditorAPI, private documents: ExtDocuments) {}
```

Comment:
By passing in `ExtDocuments` instead of just a `sourcegraph.TextDocument`, it ensures that the `document` getter returns the latest version of the document. It could pass in an `ExtDocument` that proxies that, but this code is so simple and well tested that it didn't seem necessary to do that.